appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/DMManagedConnectionFactory.java

```diff
@@ -191,34 +191,13 @@
      * @param key
      * @return
      */
-    private String getParsedKey(String key) throws ResourceException {
```

Comment:
This could return `null` for any key not starting with `"set"` as well as the `null` key but the caller did not expect to get back null ever and fixing it there didn't make much sense. Also this method already had a test checking for null or empty key in case it was removing the "set" prefix so it seem logical that passing `null` or the empty string to this should equally cause the `ResourceException`.